### DISCUSSION TITLE
How to boot LTSP from GRUB4DOS

### DISCUSSION THREAD

**Discussion by danboid** [ltsp/ltsp#629] [Q&A]
We currently boot LTSP5 / Ubuntu 16.04 using GRUB4DOS via the Windows 10 bootloader. I've mostly configured a new LTSP server running Ubuntu 20.04 but I'm not sure how to get our clients to boot off it yet.

I'm aware we won't be able to boot UEFI machines using GRUB4DOS but we'll have to try and make this config work until all of the machines get fully re-imaged, at which point we may be able to switch to ipxe if we can use that from the Windows bootloader?

Our current GRUB4DOS menu.lst looks like this:

```
default 0
timeout 0

title Ubuntu Linux
root (hd0,0)
kernel /LTSP/ipxe.lkrn
initrd /LTSP/boot.ipxe
```

What might the modern equivalent of this config look like? I'd guess that ipxe.lkrn has been replaced by ltsp.img and that boot.ipxe has been replaced by ltsp.ipxe?

**Comment by alkisg** [comment]
Just remove your initrd line. A plain ipxe is enough to boot the clients. Your boot.ipxe possibly breaks things. You might also need (or not) to fetch a newer ipxe.lkrn from boot.ipxe.org/ipxe.lkrn.

Also, grub4dos was deprecated many years ago, it's usually better to use win32-loader which uses the latest grub2. We have nice instructions in Greek, you may use google translate if you like:
https://ts.sch.gr/docs/linux/guides/win32-loader/

**Reply by danboid** [reply]
I thought maybe something like:

chain http://asteria.cs.salford.ac.uk/ltsp/ltsp.ipxe

but that isn't working.

**Comment by danboid** [comment]
Our old LTSP server was configured like this:

chain http://leto.cs.salford.ac.uk/boot.php?ip=${ip}&mac=${mac}&asset=${asset}

root@leto:/var/www/html# cat boot.php 
#!ipxe
kernel http://leto.cs.salford.ac.uk/ltsp/vmlinuz-generic
initrd http://leto.cs.salford.ac.uk/ltsp/initrd.img-generic
imgargs vmlinuz-generic append ro initrd=initrd.img-generic init=/sbin/init-ltsp root=/dev/nbd0 nbdroot=146.87.15.145:/opt/ltsp/amd64
boot

I presume we'll have to do the same again.

**Reply by danboid** [reply]
If I'm recreating this boot method, I'm not sure about the

`root=/dev/nbd0 nbdroot=146.87.15.145:/opt/ltsp/amd64`

kernel arguments. Is LTSP20 still using nbd or would I need to replace this bit?

**Reply by danboid** [reply]
Looks like I should use

`root=/dev/nfs nfsroot=${srv}:/srv/ltsp ltsp.image=images/${img}.img loop.max_part=9`

instead according to ltsp.ipxe

**Comment by danboid** [comment]
I'll have to check with a network guy that all the required ports should be open.

If I do need to use a custom bootchain file again, I think it would look like this, although I've tried it and its not working:

```
#!ipxe
kernel http://asteria.cs.salford.ac.uk/ltsp/Ubuntu-20.04.raw/vmlinuz
initrd http://asteria.cs.salford.ac.uk/ltsp/ltsp.img
imgargs vmlinuz append root=/dev/nfs nfsroot=asteria.cs.salford.ac.uk:/srv/ltsp ltsp.image=images/Ubuntu-20.04.raw.img loop.max_part=9
boot
```
